### ISSUE TITLE
project setup

### ISSUE THREAD

**Issue by ravivarma-dot** [dotnet-presentations/blazor-workshop#294]
It's my first time going through a Blazor app and I found your blazor-workshop app useful.

I have cloned the repo and started working on it but could you a tutorial on how to setup the project the you did?

It felt like I missed important step of setting up the project from scratch in this tutorial.

**Comment by lohithgn** [comment]
If you go through the 00-get-started docs page here: https://github.com/dotnet-presentations/blazor-workshop/blob/master/docs/00-get-started.md - it talks about two links. They are:
- Blazor Getting Started https://aka.ms/blazor-getting-started
- Build Your First App https://docs.microsoft.com/aspnet/core/tutorials/build-your-first-blazor-app

That forms the basis of working with this workshop. You will need to first make sure that you have finished the above two guides before starting on this workshop. Once you understand the 2 guides - then follow along with this workshop. To speed up things - the workshop comes with a starting point and the starting point has already set up the projects for you. Typically this is how most workshop guidelines will be i.e. they provide a starting point to you.

Hope this helps.

note: If this answers your question do consider closing this issue. Thanks.

**Comment by woodya-coder** [comment]
I felt the same way. The pizza project has four projects: client, shared, server and something else. The above link only creates a single project.  How did you end up with four projects?  I also don't understand how you got to the beginning starting point.

**Comment by scout208** [comment]
It would be nice to have some more explanation as to why we are using four projects instead of one. I'm kind of confused on that as well.

**Comment by ylr-research** [comment]
When you create a Blazor Hosted solution or what people commonly refer as Blazor Server-Side, you end up with 3 projects. Client, Server and Shared (mostly shared classes between Client and Server. I.e., models). ComponentsLibrary project it's just another project containing reusable razor components.

**Comment by scout208** [comment]
@ylr-research Which template are you using to create the solution? Every time I've created a Blazor Server app it has created only one project. Are you referring to the project structure created when you select ASP.NET Core Hosted option when creating a Blazor WebAssembly project?

**Comment by ylr-research** [comment]
@scout208 yes.

**Comment by scout208** [comment]
@ylr-research Oh, I see. I was confused by the difference between Blazor Hosted (which is part of the WASM template) and Blazor Server (which has it's own template).

**Comment by ylr-research** [comment]
I think this issue has been answered. Can this be closed @ravivarma-dot?
